discord_connector.py

```diff
@@ -309,6 +383,86 @@ def run_in_loop(self, coroutine, action_result, error_message=""):
             return action_result.set_status(phantom.APP_ERROR,
                                             f"Other exception. Error type: {e.__class__.__name__} Details: {str(e)}"), None
 
+    def _handle_on_poll(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        container_count = param.get('container_count', None)
+
+        status, channels = self.run_in_loop(self._guild.fetch_channels(), action_result,
+                                            error_message="Cannot fetch channel from Discord.")
+        if not status:
+            self.save_progress("action result: Cannot Poll messages, unable to fetch channels ")
+            return action_result.set_status(phantom.APP_ERROR, "action result: Cannot Poll messages")
+
+        self._status = self.load_state()
+        last_poll_date = self._state.get("last_poll_date", None)
+        if last_poll_date is not None:
+            last_poll_date_parsing_status, last_poll_date = self.parse_date(last_poll_date)
+            if not last_poll_date_parsing_status:
+                return action_result.set_status(phantom.APP_ERROR, "Unable to parse last poll date")
+
+        self.save_progress("lading last poll date: {}".format(last_poll_date))
+
+        newest_message = last_poll_date
+        if newest_message is None:
+            newest_message = datetime.min.replace(tzinfo=pytz.UTC)
```

Comment:
```suggestion
        newest_message = last_poll_date or datetime.min.replace(tzinfo=pytz.UTC)
```
Basically you know that `last_poll_date` is False if it's none, so this can be written like this without any extra assignments upfront. Or at least, you can use the walrus notation for the condition.

Reply:
fair point